File: src/adapter/festa/fantasia/RoupaDeFantasma.java
package adapter.festa.fantasia;

// Adaptador especializado

public class RoupaDeFantasma extends Fantasia {

	public RoupaDeFantasma(Pessoa pessoa) {
		super(pessoa);
	}

	@Override
	public String nomeDoPersonagem() {
		return "O Fantasma " + pessoa.obterNome();
	}

	@Override
	public String descricaoDaRoupa() {
		return "Um lençol com furos para os olhos";
	}

	@Override
	public String bebidaFavorita() {
		return pessoa.obterBebidaFavorita();
	}
}

File: src/adapter/festa/fantasia/RoupaDeFantasma.kt
package adapter.festa.fantasia

// Adaptador especializado

class RoupaDeFantasma(pessoa: Pessoa) : Fantasia(pessoa) {

    override fun nomeDoPersonagem(): String {
        return "O Fantasma " + pessoa.obterNome()
    }

    override fun descricaoDaRoupa(): String {
        return "Um lençol com furos para os olhos"
    }

    override fun bebidaFavorita(): String {
        return pessoa.obterBebidaFavorita()
    }
}
